tests/fields/test_dict_field.py

```diff
@@ -43,13 +44,14 @@ class BlogPost(Document):
         with pytest.raises(ValidationError):
             post.validate()
 
-        post.info = {"the.title": "test"}
-        with pytest.raises(ValidationError):
-            post.validate()
+        if get_mongodb_version() < MONGODB_36:
+            post.info = {"the.title": "test"}
+            with pytest.raises(ValidationError):
+                post.validate()
 
-        post.info = {"nested": {"the.title": "test"}}
-        with pytest.raises(ValidationError):
-            post.validate()
+            post.info = {"nested": {"the.title": "test"}}
+            with pytest.raises(ValidationError):
+                post.validate()
```

Comment:
It is missing a test that checks that dict with dot AND dollars can be stored (when MONGO >= 3.6)